### PR TITLE
[sgd] Fixes TrainingOperator wrap model incorrect

### LINKED ISSUE #14352: [sgd] TrainingOperator wrap model incorrect
### What is the problem?

```python
---------------------------------------------------------------------------
RayTaskError(AttributeError)              Traceback (most recent call last)
<ipython-input-12-3435b262f1ae> in <module>
----> 1 trainer.train()

~/miniconda3/envs/ray/lib/python3.7/site-packages/ray-2.0.0.dev0-py3.7-linux-x86_64.egg/ray/util/sgd/torch/torch_trainer.py in train(self, num_steps, profile, reduce_results, max_retries, info, dataset)
    413             self._resize_worker_group()
    414         success, worker_stats = self.worker_group.train(
--> 415             num_steps=num_steps, profile=profile, info=info, dataset=dataset)
    416         # Fault handling
    417         for i in range(max_retries):

~/miniconda3/envs/ray/lib/python3.7/site-packages/ray-2.0.0.dev0-py3.7-linux-x86_64.egg/ray/util/sgd/torch/worker_group.py in train(self, num_steps, profile, info, dataset)
    323         remote_worker_stats = self._train(num_steps, profile, info, dataset)
    324         # Check if each worker has failed before calling ray.get.
--> 325         success = check_for_failure(remote_worker_stats)
    326         if success:
    327             return success, ray.get(remote_worker_stats)

~/miniconda3/envs/ray/lib/python3.7/site-packages/ray-2.0.0.dev0-py3.7-linux-x86_64.egg/ray/util/sgd/utils.py in check_for_failure(remote_values)
    242         while len(unfinished) > 0:
    243             finished, unfinished = ray.wait(unfinished)
--> 244             finished = ray.get(finished)
    245         return True
    246     except RayActorError as exc:

~/miniconda3/envs/ray/lib/python3.7/site-packages/ray-2.0.0.dev0-py3.7-linux-x86_64.egg/ray/_private/client_mode_hook.py in wrapper(*args, **kwargs)
     45         if client_mode_should_convert():
     46             return getattr(ray, func.__name__)(*args, **kwargs)
---> 47         return func(*args, **kwargs)
     48
     49     return wrapper

~/miniconda3/envs/ray/lib/python3.7/site-packages/ray-2.0.0.dev0-py3.7-linux-x86_64.egg/ray/worker.py in get(object_refs, timeout)
   1426                     worker.core_worker.dump_object_store_memory_usage()
   1427                 if isinstance(value, RayTaskError):
-> 1428                     raise value.as_instanceof_cause()
   1429                 else:
   1430                     raise value

RayTaskError(AttributeError): ray::DistributedTorchRunner.train_epoch() (pid=4620, ip=10.239.10.106)
  File "python/ray/_raylet.pyx", line 497, in ray._raylet.execute_task
  File "python/ray/_raylet.pyx", line 444, in ray._raylet.execute_task.function_executor
  File "/home/lxy/miniconda3/envs/ray/lib/python3.7/site-packages/ray-2.0.0.dev0-py3.7-linux-x86_64.egg/ray/util/sgd/torch/distributed_torch_runner.py", line 112, in train_epoch
    num_steps=num_steps, profile=profile, info=info, iterator=iterator)
  File "/home/lxy/miniconda3/envs/ray/lib/python3.7/site-packages/ray-2.0.0.dev0-py3.7-linux-x86_64.egg/ray/util/sgd/torch/torch_runner.py", line 140, in train_epoch
    train_stats = self.training_operator.train_epoch(iterator, info)
  File "/home/lxy/miniconda3/envs/ray/lib/python3.7/site-packages/ray-2.0.0.dev0-py3.7-linux-x86_64.egg/ray/util/sgd/torch/training_operator.py", line 504, in train_epoch
    model.train()
AttributeError: 'list' object has no attribute 'train'
```

*Ray version and other system information (Python version, TensorFlow version, OS):*

### Reproduction (REQUIRED)
Please provide a short code snippet (less than 50 lines if possible) that can be copy-pasted to reproduce the issue. The snippet should have **no external library dependencies** (i.e., use fake or mock data / environments):

```python
In [1]: import ray
In [2]: from ray.util.sgd.torch.training_operator import TrainingOperator
In [3]: from ray.util.sgd.torch import TorchTrainer
In [4]: from ray.util.sgd.torch.examples.train_example import data_creator, LinearDataset
In [5]: import torch
In [6]: ray.init()
2021-02-25 23:21:36,997 INFO services.py:1226 -- View the Ray dashboard at http://127.0.0.1:8265
Out[6]:
{'node_ip_address': '10.239.10.106',
 'raylet_ip_address': '10.239.10.106',
 'redis_address': '10.239.10.106:6379',
 'object_store_address': '/tmp/ray/session_2021-02-25_23-21-35_660533_4478/sockets/plasma_store',
 'raylet_socket_name': '/tmp/ray/session_2021-02-25_23-21-35_660533_4478/sockets/raylet',
 'webui_url': '127.0.0.1:8265',
 'session_dir': '/tmp/ray/session_2021-02-25_23-21-35_660533_4478',
 'metrics_export_port': 63812,
 'node_id': 'd5cef3449cd7b161fd6b50235207dd928ef4eaa744b3e1b9f34f0af1'}

In [7]: model = torch.nn.Sequential(torch.nn.Linear(1, 1))
In [8]: optimizer = torch.optim.SGD(model.parameters(), lr=0.1)
In [9]: criterion = torch.nn.MSELoss
In [10]: class Operator(TrainingOperator):
    ...:     def setup(self, config):
    ...:         self.model, self.optimizer, self.criterion = self.register(models=model, optimizers=optimizer, criterion=criterion)
    ...:         train_ld, val_ld = data_creator(config)
    ...:         self.register_data(train_loader=train_ld, validation_loader=val_ld)
    ...:
In [11]: trainer = TorchTrainer(training_operator_cls=Operator, num_workers=2, use_local=False, use_gpu=False)
In [12]: trainer.train()
```

If the code snippet cannot be run by itself, the issue will be closed with "needs-repro-script".

- [ ] I have verified my script runs in a clean environment and reproduces the issue.
- [ ] I have verified the issue also occurs with the [latest wheels](https://docs.ray.io/en/master/installation.html).

### PR DESCRIPTION
**Pull request by ConeyLiu**
## Related issue number

Closes #14352 

## Checks

- [x] I've run `scripts/format.sh` to lint the changes in this PR.
- [ ] I've included any doc changes needed for https://docs.ray.io/en/master/.
- [ ] I've made sure the tests are passing. Note that there might be a few flaky tests, see the recent failures at https://flakey-tests.ray.io/
- Testing Strategy
   - [x] Unit tests
   - [ ] Release tests
   - [ ] This PR is not tested :(

### CODE REVIEW

**Comment by ConeyLiu** [comment]
cc @richardliaw

**Comment by amogkam** [comment]
Hey @ConeyLiu thanks a bunch for this PR! This is a good catch- `torch.nn.Sequential` is an `Iterable` so we need to account for that.

There are some existing SGD tests that are failing right now. Can you take a look and fix those? Thanks!

**Comment by ConeyLiu** [comment]
Sure, I will take a look.

**Comment by ConeyLiu** [comment]
The following lint errors seem to break sphinx build. I have no idea to fix it. I have tried different sphinx versions, the error still occurs.
```
Traceback (most recent call last):
  File "/opt/miniconda/lib/python3.7/site-packages/sphinx/ext/autodoc/importer.py", line 32, in import_module
    return importlib.import_module(modname)
  File "/opt/miniconda/lib/python3.7/importlib/__init__.py", line 127, in import_module
    return _bootstrap._gcd_import(name[level:], package, level)
  File "<frozen importlib._bootstrap>", line 1006, in _gcd_import
  File "<frozen importlib._bootstrap>", line 983, in _find_and_load
  File "<frozen importlib._bootstrap>", line 967, in _find_and_load_unlocked
  File "<frozen importlib._bootstrap>", line 677, in _load_unlocked
  File "<frozen importlib._bootstrap_external>", line 728, in exec_module
  File "<frozen importlib._bootstrap>", line 219, in _call_with_frames_removed
  File "/home/travis/build/ray-project/ray/python/ray/util/sgd/torch/lightning_operator.py", line 29, in <module>
    TrainerOptimizersMixin):
TypeError: metaclass conflict: the metaclass of a derived class must be a (non-strict) subclass of the metaclasses of all its bases
```

File: doc/source/raysgd/raysgd_ref.rst [hunk 1]
```diff
@@ -25,7 +25,7 @@ PyTorch TrainingOperator
 CreatorOperator
 ~~~~~~~~~~~~~~~~
 
-.. autoclass:: ray.util.sgd.torch.training_operator.CreatorOperator
```
Comment (ConeyLiu) [inline doc/source/raysgd/raysgd_ref.rst:28]: This breaks the lint check too.

**Comment by ConeyLiu** [comment]
cc @amogkam

**Review by amogkam — APPROVED** [review]
Ah sorry this slipped- thanks for the ping and for the PR. Lgtm, we can merge once CI finishes.

**Comment by ConeyLiu** [comment]
thanks @amogkam

**Comment by amogkam** [comment]
@richardliaw can you merge this please

**Pull request merged**